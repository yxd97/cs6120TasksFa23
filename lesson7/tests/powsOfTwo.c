#include <stdio.h>
#include <stdlib.h>
// ARGS: 35
int main(int argc, char** argv) {
    if (argc < 2) {
        return 1;
    }
    int x = atoi(argv[1]);
    int r1 = x * 2;
    printf("%d x %d = %d\n", x, 2, r1);
    int r2 = x * 16;
    printf("%d x %d = %d\n", x, 16, r2);
    int r3 = x * 65536;
    printf("%d x %d = %d\n", x, 65536, r3);
    int r4 = x * 1024;
    printf("%d x %d = %d\n", x, 1024, r4);
    int r5 = x * 1;
    printf("%d x %d = %d\n", x, 1, r5);
    return 0;
}
