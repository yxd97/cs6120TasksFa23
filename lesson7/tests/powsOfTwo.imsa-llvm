; ModuleID = 'tests/powsOfTwo.c'
source_filename = "tests/powsOfTwo.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [14 x i8] c"%d x %d = %d\0A\00", align 1

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main(i32 noundef %0, ptr noundef %1) #0 {
  %3 = alloca i32, align 4
  %4 = alloca i32, align 4
  %5 = alloca ptr, align 8
  %6 = alloca i32, align 4
  %7 = alloca i32, align 4
  %8 = alloca i32, align 4
  %9 = alloca i32, align 4
  %10 = alloca i32, align 4
  %11 = alloca i32, align 4
  store i32 0, ptr %3, align 4
  store i32 %0, ptr %4, align 4
  store ptr %1, ptr %5, align 8
  %12 = load i32, ptr %4, align 4
  %13 = icmp slt i32 %12, 2
  br i1 %13, label %14, label %15

14:                                               ; preds = %2
  store i32 1, ptr %3, align 4
  br label %55

15:                                               ; preds = %2
  %16 = load ptr, ptr %5, align 8
  %17 = getelementptr inbounds ptr, ptr %16, i64 1
  %18 = load ptr, ptr %17, align 8
  %19 = call i32 @atoi(ptr noundef %18) #3
  store i32 %19, ptr %6, align 4
  %20 = load i32, ptr %6, align 4
  %21 = mul nsw i32 %20, 2
  %22 = shl i32 %20, 1
  %23 = add i32 0, %22
  store i32 %23, ptr %7, align 4
  %24 = load i32, ptr %6, align 4
  %25 = load i32, ptr %7, align 4
  %26 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %24, i32 noundef 2, i32 noundef %25)
  %27 = load i32, ptr %6, align 4
  %28 = mul nsw i32 %27, 16
  %29 = shl i32 %27, 4
  %30 = add i32 0, %29
  store i32 %30, ptr %8, align 4
  %31 = load i32, ptr %6, align 4
  %32 = load i32, ptr %8, align 4
  %33 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %31, i32 noundef 16, i32 noundef %32)
  %34 = load i32, ptr %6, align 4
  %35 = mul nsw i32 %34, 65536
  %36 = shl i32 %34, 16
  %37 = add i32 0, %36
  store i32 %37, ptr %9, align 4
  %38 = load i32, ptr %6, align 4
  %39 = load i32, ptr %9, align 4
  %40 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %38, i32 noundef 65536, i32 noundef %39)
  %41 = load i32, ptr %6, align 4
  %42 = mul nsw i32 %41, 1024
  %43 = shl i32 %41, 10
  %44 = add i32 0, %43
  store i32 %44, ptr %10, align 4
  %45 = load i32, ptr %6, align 4
  %46 = load i32, ptr %10, align 4
  %47 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %45, i32 noundef 1024, i32 noundef %46)
  %48 = load i32, ptr %6, align 4
  %49 = mul nsw i32 %48, 1
  %50 = shl i32 %48, 0
  %51 = add i32 0, %50
  store i32 %51, ptr %11, align 4
  %52 = load i32, ptr %6, align 4
  %53 = load i32, ptr %11, align 4
  %54 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %52, i32 noundef 1, i32 noundef %53)
  store i32 0, ptr %3, align 4
  br label %55

55:                                               ; preds = %15, %14
  %56 = load i32, ptr %3, align 4
  ret i32 %56
}

; Function Attrs: nounwind willreturn memory(read)
declare i32 @atoi(ptr noundef) #1

declare i32 @printf(ptr noundef, ...) #2

attributes #0 = { noinline nounwind optnone uwtable "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #1 = { nounwind willreturn memory(read) "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #2 = { "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #3 = { nounwind willreturn memory(read) }

!llvm.module.flags = !{!0, !1, !2, !3, !4}
!llvm.ident = !{!5}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{i32 8, !"PIC Level", i32 2}
!2 = !{i32 7, !"PIE Level", i32 2}
!3 = !{i32 7, !"uwtable", i32 2}
!4 = !{i32 7, !"frame-pointer", i32 2}
!5 = !{!"Ubuntu clang version 17.0.2 (++20230929073225+098e653a5bed-1~exp1~20230929073241.48)"}
