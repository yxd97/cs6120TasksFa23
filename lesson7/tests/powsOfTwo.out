35 x 2 = 70
35 x 16 = 560
35 x 65536 = 2293760
35 x 1024 = 35840
35 x 1 = 35
